src/Dfc.CourseDirectory.WebV2/MultiPageTransaction/InvalidMptxInstanceContextActionFilter.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Dfc.CourseDirectory.WebV2.MultiPageTransaction
+{
+    public class InvalidMptxInstanceContextActionFilter : IActionFilter
+    {
+        private static readonly Type _instanceContextType = typeof(MptxInstanceContext<>);
+        private static readonly Type _instanceContextWithParentType = typeof(MptxInstanceContext<,>);
+
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+        }
+
+        public void OnActionExecuting(ActionExecutingContext context)
```

Comment:
Suggestion: the contents of this method is a little hard for me to parse, particularly being less familiar with action filters; it might be good to break out a couple of tiny well named methods to make it clearer what it's doing. The use of comment might also be a hint that this is needed. I think that the comment could possibly be completely replaced by good method call names which would be a small win.